client/rpc/rpc_sender.go

```diff
@@ -98,13 +96,14 @@ func (s *Sender) Send(_ context.Context, call proto.Call) {
 			// there's visiblity that this is happening. Some of the errors
 			// we'll sweep up in this net shouldn't be retried, but we can't
 			// really know for sure which.
-			log.Warningf("failed to send RPC request %s: %v", method, c.Error)
-			return retry.Continue, nil
+			log.Warningf("failed to send RPC request %s: %v", method, err)
+			continue
 		}
 
 		// On successful post, we're done with retry loop.
-		return retry.Break, nil
-	}); err != nil {
+		break
```

Comment:
i think i prefer the current thing. there's already a nil check for `err` down below. six of one...